sdk/trace/batch_span_processor.go

```diff
@@ -166,32 +167,37 @@ func WithBlocking() BatchSpanProcessorOption {
 	}
 }
 
-func (bsp *BatchSpanProcessor) processQueue() {
-	batch := make([]*export.SpanData, 0, bsp.o.MaxExportBatchSize)
+// processQueue removes spans from the `queue` channel until there is
+// no more data.  It calls the exporter in batches of up to
+// MaxExportBatchSize until all the available data has been processed
+// or the stop channel closes.
+func (bsp *BatchSpanProcessor) processQueue(batch *[]*export.SpanData) {
 	for {
-		var sd *export.SpanData
-		var ok bool
-		select {
-		case sd = <-bsp.queue:
-			if sd != nil && sd.SpanContext.IsSampled() {
-				batch = append(batch, sd)
+		// Read spans until either the buffer fills or the
+		// queue is empty.
+		ok := true
+		for ok && len(*batch) < bsp.o.MaxExportBatchSize {
+			var sd *export.SpanData
+			select {
```

Comment:
I think that the `<-bsp.stopCh` case can be simply dropped. The loop that invokes this function checks one of the two channels - `bsp.stopCh` and a ticker. We can add a check inside the ticker case if the stop channel was closed and return.

Reply:
OK, done. I was worrying about the case where there is a never-ending stream of spans, in which case this will never return. I think it's ok behavior this way.